nodejs/week1/src/backend/index.js

```diff
@@ -0,0 +1,47 @@
+const express = require("express");
+const meals = require("./data/meals.json");
+const reservations = require("./data/reservations.json");
+
+const app = express();
+
+app.get("/meals", (req, res)=>{
+    res.json(meals);
+});
+
+app.get("/reservations", (req, res)=>{
+    res.json(reservations);
+});
+
+app.get("/cheap-meals", (req, res)=>{
+    cheapMeals = meals.filter((cheapmeal)=>{
+        return cheapmeal.price < 65;
+    });
+    res.json(cheapMeals);
+});
+
+app.get("/large-meals", (req, res)=>{
+    largeMeals = meals.filter((largermeal)=>{
+        return largermeal.maxNumberOfGuests>7;
+    });
+    res.json(largeMeals)
+});
+
+app.get("/meals", (req, res)=>{
+    const randommeal = meals[Math.floor(Math.random() * meals.length)];
+    res.json(randommeal);
+});
+
+app.get("/reservations", (req, res)=>{
+    const randomReservation = 
+    reservations[Math.floor(Math.random() * reservations.length)];
```

Comment:
Here you're using the same logic than you used above to get a random integer between 0 and N where N is the length of an array. When you find yourself writing the same code more than twice, consider whether you can extract a helper function that you can reuse from both places. This principle is commonly known as DRY (Don't Repeat Yourself). Keeping your code DRY helps you right more maintainable code because:

- If there's a bug in this code, it's only in one place and you only need to fix it once
- When you create a helper function other team mates can benefit from it and reuse your code
- When someone looks at your code they need to read it slowly and understand what is going on. When they read a helper function like `getRandomInteger(maxValue)` they automatically understand what the function does without needing to  exert their cognitive abilities a lot.